client/my-sites/themes/controller.js

```diff
@@ -49,27 +45,43 @@ function getProps( context ) {
 	};
 
 	return {
-		title,
 		tier,
 		search: context.query.s,
 		trackScrollPage: boundTrackScrollPage,
 		runClientAnalytics
 	};
-}
+};
 
-function makeElement( ThemesComponent, Head, store, props ) {
-	return(
-		<ReduxProvider store={ store }>
-			<Head title={ props.title } tier={ props.tier || 'all' }>
-				<ThemesComponent { ...omit( props, [ 'title', 'runClientAnalytics' ] ) } />
+const LoggedInHead = ( { children, context: { params: { tier, site_id: siteID } } } ) => (
+	<Head
+		title={ buildTitle(
+			i18n.translate( 'Themes', { textOnly: true } ),
+			{ siteID }
+		) }
+		tier={ tier || 'all' }>
+		{ children }
+	</Head>
+);
+
+// This is generic -- nothing themes-specific in here!
+function makeElement( Component, getProps, Head, sideEffects = function() {} ) {
+	// How do we handle the dispatch?
+	return ( context, next ) => {
```

Comment:
I see. Question: if `Head` is to extract relevant props from `context` instead of from the output of `getProps`, this would likely make SSR caching less evident [^1]. How would you deal with that in this scenario?

[^1]: Since the idea would be to cache as a function of `( route, getProps() )`, roughly speaking. That is, `getProps` would be the one selector for a specific route.